test/Npgsql.Tests/CommandTests.cs

```diff
@@ -158,18 +158,37 @@ public async Task Timeout()
 
             // Mono throws a socket exception with WouldBlock instead of TimedOut (see #1330)
             var isMono = Type.GetType("Mono.Runtime") != null;
-            using (var conn = await OpenConnectionAsync(ConnectionString + ";CommandTimeout=1"))
-            using (var cmd = CreateSleepCommand(conn, 10))
-            {
-                Assert.That(() => cmd.ExecuteNonQuery(), Throws.Exception
-                    .TypeOf<NpgsqlException>()
-                    .With.InnerException.TypeOf<IOException>()
-                    .With.InnerException.InnerException.TypeOf<SocketException>()
-                    .With.InnerException.InnerException.Property(nameof(SocketException.SocketErrorCode)).EqualTo(isMono ? SocketError.WouldBlock : SocketError.TimedOut)
-                    );
-                Assert.That(conn.FullState, Is.EqualTo(ConnectionState.Broken));
-            }
+            using var conn = await OpenConnectionAsync(ConnectionString + ";CommandTimeout=1");
+            using var cmd = CreateSleepCommand(conn, 10);
+            Assert.That(() => cmd.ExecuteNonQuery(), Throws.Exception
+                .TypeOf<NpgsqlException>()
+                .With.InnerException.TypeOf<IOException>()
+                .With.InnerException.InnerException.TypeOf<SocketException>()
+                .With.InnerException.InnerException.Property(nameof(SocketException.SocketErrorCode)).EqualTo(isMono ? SocketError.WouldBlock : SocketError.TimedOut)
+                );
+            Assert.That(conn.FullState, Is.EqualTo(ConnectionState.Broken));
+        }
+
+// Timeout for async queries is not supported for .net 4.6.1
```

Comment:
I don't see any other `#if` in your PR.
What happens in .NET Framework 4.6.1?
Is it just silently ignored?

Reply:
From [references](https://referencesource.microsoft.com/#mscorlib/system/io/stream.cs,e224b4bec8748849) for .net 4.8 (as I remember, it was the same for .net 4.6.1), `NetworkStream` does not override `Stream.ReadAsync` and `Stream.WriteAsync`. And implementation from `Stream` checks `CancellationToken` only once, essentially ignoring it. So, for .net 4.6.1 there is no way to cancel `NetworkStream.ReadAsync` and `NetworkStream.WriteAsync` through `CancellationToken`.

Reply:
Yeah, actual cancellation token handling for NetworkStream was added in .NET Core 3.0 (I think). It's reasonable to compiled this out.